cmd/kwil-cli/cmds/utils/dbid.go

```diff
@@ -0,0 +1,76 @@
+package utils
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/kwilteam/kwil-db/cmd/common/display"
+	"github.com/kwilteam/kwil-db/core/utils"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dbidLong = `"dbid" generates a dbid for a given schema name and deployer.`
+
+	dbidExample = `# Generate a dbid for a schema and deployer
+kwil-cli utils dbid --schema=myschema --deployer=0x1234567890abcdef
+
+# Maintain the exact deployer address, and do not trim off the 0x prefix
+kwil-cli utils dbid --schema=myschema --deployer=0xnot_an_eth_address --no-trim`
+)
+
+func dbidCmd() *cobra.Command {
+	var schema, deployer string
+	var noTrim bool
+
+	cmd := &cobra.Command{
+		Use:     "generate-dbid",
+		Short:   dbidLong,
+		Long:    dbidLong,
+		Example: dbidExample,
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			deployerBts := []byte(deployer)
+			if strings.HasPrefix(deployer, "0x") && !noTrim {
+				deployer = deployer[2:]
+				var err error
+				deployerBts, err = hex.DecodeString(deployer)
+				if err != nil {
+					return display.PrintErr(cmd, fmt.Errorf(`deployer address "%s" has 0x prefix but is not a valid hex string. try using the --no-trim flag`, deployer))
+				}
+			}
+
+			dbid := utils.GenerateDBID(schema, deployerBts)
+			return display.PrintCmd(cmd, &dbidOutput{DBID: dbid})
+		},
+	}
+
+	cmd.Flags().StringVar(&schema, "schema", "", "Schema name")
+	cmd.Flags().StringVar(&deployer, "deployer", "", "Deployer address")
+	cmd.Flags().BoolVar(&noTrim, "no-trim", false, "Maintain the exact deployer address")
+	// mark required flags
+	cmd.MarkFlagRequired("schema")
+	cmd.MarkFlagRequired("deployer")
+
+	return cmd
+}
+
+type dbidOutput struct {
+	DBID string `json:"dbid"`
+}
+
+func (d *dbidOutput) MarshalJSON() ([]byte, error) {
+	// Alias is used to avoid infinite recursion when calling json.Marshal
+	type Alias dbidOutput
+	return json.Marshal(&struct {
+		Alias
+	}{
+		Alias: (Alias)(*d),
+	})
```

Comment:
I'm pretty certain that the anonymous struct embedding `Alias` isn't needed.

```go
	type Alias dbidOutput
	return json.Marshal((Alias)(*d))
```

```go
func Example_dbidOutput_json() {
	display.Print(&dbidOutput{DBID: "asdf"}, nil, "json")
	// Output:
	// {
	//   "result": {
	//     "dbid": "asdf"
	//   },
	//   "error": ""
	// }
}
```

Reply:
I tested it without it, and I hit an overflow. It is because It is implementing a `MarshalJSON` method, so calling `json.Marshal(d)` would be infinitely recursive.

Reply:
Right, I understand the recursion issue needs Alias, but you have this anonymous struct in addition to alias.  Can you try with just this?

```go
	type Alias dbidOutput
	return json.Marshal((Alias)(*d))
```

That works for me.  The alias is sufficient to drop the methods of the underlying type.

Reply:
Oh whoops